bleak/backends/scanner.py

```diff
@@ -3,44 +3,67 @@
 import inspect
 import os
 import platform
-from typing import Awaitable, Callable, Dict, List, Optional, Tuple, Type
+from typing import (
+    Any,
+    Awaitable,
+    Callable,
+    Dict,
+    List,
+    NamedTuple,
+    Optional,
+    Tuple,
+    Type,
+)
 
 from ..exc import BleakError
 from .device import BLEDevice
 
 
-class AdvertisementData:
+class AdvertisementData(NamedTuple):
```

Comment:
I wonder if it would be faster as a dataclass?

Reply:
Nope. Named tuple is faster

Reply:
Found this while digging around https://medium.com/@jacktator/dataclass-vs-namedtuple-vs-object-for-performance-optimization-in-python-691e234253b9

But will get better data when I run in through the profiler since we will get actual use case data vs theoretical

Reply:
Part of the reason for using NamedTuple was for the immutability, but if there is a way to speed it up, we can consider alternatives.

Reply:
Makes sense.  I don't think it's going to be an issue. Will know for sure once I do the profiles on the production systems

Reply:
Don't forget to add `frozen=True` to dataclasses when profiling 👍